src/ocean/net/ssl/SslClientConnection.d

```diff
@@ -0,0 +1,622 @@
+
+        Params:
+            host_name = the host which was connected to
+            func_name = name of the method that failed
+            msg = Pointer to a C string desribing the error
+            file = file where exception is thrown
+            line = line where exception is thrown
+
+    *******************************************************************/
+
+    public void ssl_error ( cstring host_name, istring func_name,
+            Const!(char*) c_msg, istring file = __FILE__, long line = __LINE__ )
+    {
+        throw this.exception.set(host_name, func_name, StringC.toDString(c_msg),
+            file, line);
+    }
+}
+
+
+
+/*******************************************************************************
+
+    Initializes SSL and creates a global SSL_CTX object
+
+    This function must be called before an SSL clients can be created
+
+    Params:
+        ca_path = a directory containing CA certificates in PEM format
+        ca_file = pointer to a file of CA certificates in PEM format, or
+                null. The file can containe several CA certificates.
+
+    Returns:
+
+        0 if successful, otherwise returns an error code
+
+*******************************************************************************/
+
+
+public ulong initializeSslAndCreateCtx ( Const!(char *) ca_path,
+    Const!(char *) ca_file = null )
+{
+    globalSslContext = null;
+
+    // Initialize SSL
+
+    SSL_library_init();
+
+    // Load  both libssl and libcrypto strings
+
+    SSL_load_error_strings();
+
+
+    // Load the SSL v2 or v3 function table
+
+    auto method = SSLv23_method();
+
+    if ( !method )
+    {
+        return ERR_get_error();
+    }
+
+    // Create an SSL context to use for all future SSL operations
+
+    globalSslContext = SSL_CTX_new(method);
+
+    if ( !globalSslContext )
+    {
+        return ERR_get_error();
+    }
+
+    SSL_CTX_set_verify(globalSslContext, SSL_VERIFY_PEER, null);
+
+    SSL_CTX_set_verify_depth(globalSslContext, 5);
+
+    // Remove the most problematic options. Because SSLv2 and SSLv3 have been
+    // removed, a TLSv1.0 handshake is used. Clients created from this context
+    // will accept TLSv1.0 and above. An added benefit of TLS 1.0 and above
+    // are TLS extensions like Server Name Indicatior (SNI).
+    long flags = SSL_OP_ALL | SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3
+        | SSL_OP_NO_COMPRESSION;
+    long old_opts = SSL_CTX_set_options(globalSslContext, flags);
+
+    // I found that SSL_get_verify_results returns error 20
+    // ("unable to get local issuer certificate")
+    // unless it has the path to the directory where CA keys are stored
+    auto res = SSL_CTX_load_verify_locations(globalSslContext, ca_file,
+        ca_path);
+
+    if ( res == 1)
+    {
+        // Success
+
+        return 0;
+    }
+
+    // Technically, this isn't a fatal error, but many subsequent operations
+    // will fail
+
+    return res;
+}
+
+
+
+unittest
```

Comment:
Add `///` to generate documentation from this unittest.

Reply:
Does `/***` generate a documented unittest as well? The [`language spec`](https://dlang.org/spec/unittest.html#documented-unittests) is unclear on what exactly is required.

Reply:
It should.